cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/MaintenancePlanSerdeTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.detector;
+
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class MaintenancePlanSerdeTest {
+  public static final int TEST_BROKER_ID = 42;
+  public static final String TOPIC_NAME = "TestTopic";
+
+  @Test
+  public void testDemotePlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    DemoteBrokerPlan demoteBrokerPlanA = new DemoteBrokerPlan(System.currentTimeMillis(), TEST_BROKER_ID, Collections.singleton(TEST_BROKER_ID));
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, demoteBrokerPlanA);
+    DemoteBrokerPlan demoteBrokerPlanB = (DemoteBrokerPlan) maintenancePlanSerde.deserialize(TOPIC_NAME, serializedJson);
+    Assert.assertNotNull(demoteBrokerPlanB);
+    Assert.assertEquals(demoteBrokerPlanA.getCrc(), demoteBrokerPlanB.getCrc());
```

Comment:
[Not introduced in this patch] I noticed that `MaintenancePlanWithBrokers#_brokers`, which is used by (1) `AddBrokerPlan` (2) `DemoteBrokerPlan` and (3) `RemoveBrokerPlan` , is a `Set`. But, since it is not a `SortedSet`, I suspect that `getCrc` might potentially return a different response in case `_brokers` are iterated in different order in that function.

Can we
1. Make `MaintenancePlanWithBrokers#_brokers` a `SortedSet`, and
2. Add tests in this class that verifies the Add/Remove/Demote with multiple brokers?

Reply:
good point. done.